subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/AbstractFileSystemNodeWithChildrenTest.groovy

```diff
@@ -76,18 +80,18 @@ abstract class AbstractFileSystemNodeWithChildrenTest<NODE extends FileSystemNod
     }
 
     ChildMap<FileSystemNode> childrenWithSelectedChildReplacedBy(String replacementPath, FileSystemNode replacement) {
-        def newChildren = new ArrayList<ChildMap.Entry<FileSystemNode>>(children.entries())
+        def newChildren = new ArrayList<ChildMap.Entry<FileSystemNode>>(entriesFor(children))
```

Comment:
`Collectors.toList()` already creates an `ArrayList`, why are we passing that to a new `ArrayList`?

Reply:
Because the code then modifies the array list. `children.entries` returned something non-modifiable before.

Reply:
I mean, the way `entriesFor()` is implemented now, it already returns an `ArrayList` that you can modify.